platform-sdk/swirlds-common/src/main/java/com/swirlds/common/concurrent/internal/DefaultForkJoinWorkerThread.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.concurrent.internal;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinWorkerThread;
+
+public class DefaultForkJoinWorkerThread extends ForkJoinWorkerThread {
+
+    private final Runnable onStartup;
+
+    protected DefaultForkJoinWorkerThread(
+            @NonNull final String name,
+            @NonNull final ThreadGroup group,
+            @NonNull final ForkJoinPool pool,
+            @Nullable final Runnable onStartup) {
+        super(group, pool, true);
+        Objects.requireNonNull(name, "name must not be null");
+        Objects.requireNonNull(group, "group must not be null");
+        Objects.requireNonNull(pool, "pool must not be null");
+        setName(name);
+        this.onStartup = onStartup;
+    }
+
+    @Override
+    protected void onStart() {
+        if (onStartup != null) {
+            onStartup.run();
+        }
+        super.onStart();
```

Comment:
The documentation says that `super.onStart()` must be called in this method FIRST